### PR TITLE
feat: orderSubscriptionNow implementation

### PR DESCRIPTION
**Pull request by kibo-chandradeeptalaha**

### CODE REVIEW

File: hooks/mutations/subscription/useOrderSubscriptionNowMutation.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  const response = await client.request({
+    document: orderSubscriptionNow,
+    variables,
+  })
+
+  return response?.orderSubscriptionNow
+}
+
+export const useOrderSubscriptionNowMutation = () => {
```
Comment (TriptiVermaXP) [inline hooks/mutations/subscription/useOrderSubscriptionNowMutation.ts]: please do the documentation using typedoc

File: components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx [hunk 2]
```diff
@@ -166,14 +189,40 @@ const SubscriptionItemList = (props: SubscriptionItemListProps) => {
...
-                  <SubscriptionButton subscriptionButtonName="edit-frequency" />
-                  <SubscriptionButton subscriptionButtonName="edit-order-date" />
-                  <SubscriptionButton subscriptionButtonName="cancel-an-item" />
-                  <SubscriptionButton subscriptionButtonName="edit-billing-information" />
-                  <SubscriptionButton subscriptionButtonName="edit-shipping-address" />
-                  <SubscriptionButton subscriptionButtonName="pause-subscription" />
+                  <SubscriptionButton
+                    subscriptionButtonName="ship-an-item-now"
+                    onClickHandler={() =>
+                      handleShipItemNow({ id: subscriptionItemData?.id as string })
```
Comment (TriptiVermaXP) [inline components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx]: test case is missing

**Review by TriptiVermaXP — CHANGES_REQUESTED** [review]

File: hooks/mutations/subscription/useOrderSubscriptionNowMutation.ts (hunk 1 above)
Comment (kibo-chandradeeptalaha) [reply hooks/mutations/subscription/useOrderSubscriptionNowMutation.ts]: Updated

File: components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx (hunk 2 above)
Comment (kibo-chandradeeptalaha) [reply components/my-account/Subscription/SubscriptionItemList/SubscriptionItemList.tsx]: Added

**Review by kibo-kevinwatts — APPROVED** [review]

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]
needs hook unit tests

File: hooks/mutations/subscription/useOrderSubscriptionNow/useOrderSubscriptionNowMutation.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { useMutation, useQueryClient } from 'react-query'
```
Comment (kibo-sushant) [inline hooks/mutations/subscription/useOrderSubscriptionNow/useOrderSubscriptionNowMutation.ts:5]: missing

/**
 * @module ...
 */

Comment (kibo-chandradeeptalaha) [reply hooks/mutations/subscription/useOrderSubscriptionNow/useOrderSubscriptionNowMutation.ts:5]: Done

**Review by kibo-kevinwatts — APPROVED** [review]

**Review by TriptiVermaXP — APPROVED** [review]

**Pull request merged**